evennia/contrib/turnbattle/tb_range.py

```diff
@@ -0,0 +1,1356 @@
+
+    def func(self):
+        "This performs the actual command."
+
+        if is_in_combat(self.caller):  # If you're in combat
+            self.caller.msg("You can't rest while you're in combat.")
+            return
+
+        self.caller.db.hp = self.caller.db.max_hp  # Set current HP to maximum
+        self.caller.location.msg_contents("%s rests to recover HP." % self.caller)
+        """
+        You'll probably want to replace this with your own system for recovering HP.
+        """
+
+class CmdStatus(Command):
+    """
+    Gives combat information.
+
+    Usage:
+      status
+
+    Shows your current and maximum HP and your distance from
+    other targets in combat.
+    """
+
+    key = "status"
+    help_category = "combat"
+
+    def func(self):
+        "This performs the actual command."
+
+        combat_status_message(self.caller)
+        
+
+class CmdCombatHelp(CmdHelp):
+    """
+    View help or a list of topics
+
+    Usage:
+      help <topic or command>
+      help list
+      help all
+
+    This will search for help on commands and other
+    topics related to the game.
+    """
+    # Just like the default help command, but will give quick
+    # tips on combat when used in a fight with no arguments.
+
+    def func(self):
+        if is_in_combat(self.caller) and not self.args:  # In combat and entered 'help' alone
+            self.caller.msg("Available combat commands:|/" +
+                            "|wAttack:|n Attack an engaged target, attempting to deal damage.|/" +
+                            "|wShoot:|n Attack from a distance, if not engaged with other fighters.|/" +
+                            "|wApproach:|n Move one step cloer to a target.|/" +
+                            "|wWithdraw:|n Move one step away from a target.|/" +
+                            "|wPass:|n Pass your turn without further action.|/" +
+                            "|wStatus:|n View current HP and ranges to other targets.|/" +
+                            "|wDisengage:|n End your turn and attempt to end combat.|/")
+        else:
+            super(CmdCombatHelp, self).func()  # Call the default help command
+
+
+class BattleCmdSet(default_cmds.CharacterCmdSet):
+    """
+    This command set includes all the commmands used in the battle system.
+    """
+    key = "DefaultCharacter"
+
+    def at_cmdset_creation(self):
+        """
+        Populates the cmdset
+        """
+        self.add(CmdFight())
+        self.add(CmdAttack())
+        self.add(CmdShoot())
+        self.add(CmdRest())
+        self.add(CmdPass())
+        self.add(CmdDisengage())
+        self.add(CmdApproach())
+        self.add(CmdWithdraw())
+        self.add(CmdStatus())
+        self.add(CmdCombatHelp())
+
+
+"""
+----------------------------------------------------------------------------
+SCRIPTS START HERE
+----------------------------------------------------------------------------
+"""
+
+
+class TBRangeTurnHandler(DefaultScript):
```

Comment:
I am a little divided on placing this important component at the end of the module. The handler is used and referenced  throughout the other components, so when reading the code one wonders what it is and where it comes from. For this reason I always push for a module to be ordered so that things are defined above the code that use them. Hence I feel this should be defined higher up, or even at the top.

Reply:
I think it was my thought originally that the turn handler used lots of the other functions, it should come after them, but at some point I moved many of those functions into the turn handler itself. I'll put it higher up here and in the rest of the modules.